src/receive_imf.rs

```diff
@@ -1805,7 +1808,13 @@ async fn lookup_chat_by_reply(
 
     // If this was a private message just to self, it was probably a private reply.
     // It should not go into the group then, but into the private chat.
-    if is_probably_private_reply(context, to_ids, from_id, mime_parser, parent_chat.id).await? {
+    if parent_chat.get_type() != Chattype::Single
```

Comment:
At least if one of the contacts is verified, the new logic mustn't be triggered. If both are verified, an AEAP should happen. Otherwise it may be an attack, `Message-ID` isn't and can't be a protected header.

I even think that if both contacts are unverified, but use different keys (actually use, i.e. have mutual encryption preference and sign their messages), we shouldn't assign foreign messages to 1:1 chats.

Reply:
I'm not much into email attacks, but if it is feasible for an attacker to retrieve the msg-id of a 1:1 chat, this is indeed insecure. If both senders use the same key we could still assign them to the same chat but I think aliases are in general also allowed to have a different keys so it would not be a general solution. Maybe we should just assign to the chat if the 1:1 is not verified for now. 
We want to remove AEAP I think but we can still extend this logic if that is the case.

Reply:
Even for unverifed chats, if the contacts' keys are different, there should be a proper "Changed setup for ..." message, but it won't be present. Also there may be cases when one of the contacts doesn't have a key or doesn't sign messages with it, so the topic is more complicated than it seems.

But for AEAP, on the contrary, there's no problem because it has already happened up to this point, so we shouldn't do or check anything special.

Reply:
Suppose you're just writing with an alias you probably don't want to change your email address. Also, I don't think we want to show "Canged setup for..." because we show the "new" from-addr on the message chat bubble, so the user can see that this message originated from another sender. We don't want to unify the two senders but rather only associate the message with the correct chat. Maybe we can even think about dropping protection status in this case but you are right, this topic isn't simple and opens a lot of new questions.